src/buffer/out/textBuffer.cpp

```diff
@@ -944,6 +944,118 @@ Microsoft::Console::Render::IRenderTarget& TextBuffer::GetRenderTarget() noexcep
     return _renderTarget;
 }
 
+// Method Description:
+// - Get the COORD for the beginning of the word you are on
+// Arguments:
+// - target - a COORD on the word you are currently on
+// - wordDelimiters - what characters are we considering for the separation of words
+// - includeCharacterRun - include the character run located at the beginning of the word
+// Return Value:
+// - The COORD for the first character on the "word"  (inclusive)
+const COORD TextBuffer::GetWordStart(const COORD target, const std::wstring_view wordDelimiters, bool includeCharacterRun) const
+{
+    const auto bufferSize = GetSize();
+    COORD result = target;
+
+    // can't expand left
+    if (target.X == bufferSize.Left())
+    {
+        return result;
+    }
+
+    auto bufferIterator = GetTextDataAt(result);
+    const auto initialDelimiter = _GetDelimiterClass(*bufferIterator, wordDelimiters);
+    while (result.X > bufferSize.Left() && (_GetDelimiterClass(*bufferIterator, wordDelimiters) == initialDelimiter))
+    {
+        bufferSize.DecrementInBounds(result);
+        --bufferIterator;
+    }

```

Comment:
This may be crazy, but is this better written as a `do{}while()`?

Reply:
does making it a `do{}while` simplify any of the cases where we have to "walk off the non-class character"?

Reply:
A `do{}while()` would just cover the case where we're on a boundary. Still wouldn't cover the case above.